softlayer/vm/interface.go

```diff
@@ -8,6 +8,7 @@ import (
 )
 
 type VMCloudProperties struct {
+	VmNamePrefix             string                               `json:"vmNamePrefix,omitempty"`
```

Comment:
Well, it wasn't passing the integration test because of the VmNamePrefix/Hostname issue, which had been an issue before (it was looking for one but the other was defined so it wasn't getting passed in)
someone had fixed it by changing all the references to VmNamePrefix to Hostname, so they just need changed back. The integration tests might also fail for other reasons, but for now I'm running them after I think changing all the references in CloudProps to VmNamePrefix.  Hopefully that's all that's needed.